### PR TITLE
sync to upstream v1.13.5

### PR DESCRIPTION
**Pull request by 0xmountaintop**
merge https://github.com/ethereum/go-ethereum/commit/916d6a441a866cb618ae826c220866de118899f7 (https://github.com/ethereum/go-ethereum/releases/tag/v1.13.5)

### CODE REVIEW

**Comment by 0xmountaintop** [comment]
please don't squash merge this PR, if we want to merge, we should do "create a merge commit" for this PR
image

**Comment by colinlyguo** [comment]
this commit is not synced: https://github.com/ethereum/go-ethereum/commit/c1d5a012ea4b824e902db14e47bf147d727c2657

File: core/state_transition.go [hunk 1]
```diff
@@ -368,11 +368,11 @@ func (st *StateTransition) preCheck() error {
 			// This will panic if baseFee is nil, but basefee presence is verified
 			// as part of header validation.
 			if st.evm.Context.BaseFee != nil && msg.GasFeeCap.Cmp(st.evm.Context.BaseFee) < 0 {
-				return fmt.Errorf("%w: address %v, maxFeePerGas: %s baseFee: %s", ErrFeeCapTooLow,
```
Comment (colinlyguo) [inline core/state_transition.go:371]: the difference in the closed pr is a bit different:
image

File: params/version.go [hunk 2]
```diff
@@ -22,10 +22,10 @@ import (
 )
 
 const (
-	VersionMajor = 1          // Major version component of the current release
```
Comment (colinlyguo) [inline params/version.go:25]: should we change the version tag for release?

**Review by colinlyguo — APPROVED** [review]

**Review by georgehao — APPROVED** [review]

**Pull request merged**